contracts/partners/giveth/GivethMinter.sol

```diff
@@ -0,0 +1,211 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity ^0.8.4;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/security/ReentrancyGuard.sol";
+import "@openzeppelin/contracts/token/ERC721/IERC721.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "../../lib/strings.sol";
+
+interface IFlexiPunkTLD is IERC721 {
+
+  function owner() external view returns(address);
+
+  function mint(
+    string memory _domainName,
+    address _domainHolder,
+    address _referrer
+  ) external payable returns(uint256);
+
+}
+
+// generic punk minter contract
+// - no minting restrictions (anyone can mint)
+// - tiered pricing
+// - public good funding matching pool
+contract GivethMinter is Ownable, ReentrancyGuard {
+  address public poolAddress;
+  address public devAddress;
+
+  bool public paused = true;
+
+  uint256 public referralFee; // share of each domain purchase (in bips) that goes to the referrer
+  uint256 public tax; // share of each domain purchase (in bips) that goes to the Giveth Matching Pool
```

Comment:
@chnejohnson Let's rename this from `tax` to `pgfFee`. The word tax has a bit bad reputation with tokens 😅 

PGF stands for Public Goods Funding, and this is exactly that. A fee that goes to funding public goods via the Giveth Matching Pool. 🙂

Reply:
Right, that makes sense 😂